panacea-nft/src/types.rs

```diff
@@ -0,0 +1,29 @@
+use cosmwasm_std::Coin;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct TokenInfo {
+    pub contract: String,
+    pub description: Option<String>,
+    pub price: Coin,
+}
+
+impl TokenInfo {
+    pub fn into_token_info_with_owner(self, sender: String) -> TokenInfoWithSender {
+        TokenInfoWithSender {
+            contract: self.contract,
+            description: self.description,
+            price: self.price,
+            sender,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct TokenInfoWithSender {
+    pub contract: String,
+    pub description: Option<String>,
+    pub price: Coin,
+    pub sender: String,
```

Comment:
I'm confused what the `sender` mean. Is it different from the `minter`?

Reply:
Yes, it is different from `minter`.
When sent to another contract for NFT transaction, the owner of this NFT changes to contract.
The existing owner will be deleted, so we don't know the address to pay the MED after the transaction is completed.
In conclusion, this sender means an actor who does send_nft.

I'm still thinking about whether to use this sender as the owner of NFT or the account that sent NFT.

Reply:
Aha. Thank you. I understood. I will review all files with this perspective.